graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/CqlDirectives.java

```diff
@@ -587,6 +588,15 @@
           .validLocation(ARGUMENT_DEFINITION)
           .build();
 
+  private static final GraphQLDirective TIMESTAMP_DIRECTIVE =
+      newDirective()
+          .name(TIMESTAMP)
+          .description(
+              "Annotates a parameter to indicate that it will be used as a write timestamp for this row."
+                  + "This is only allowed for insert and update mutations. It must be a long value.")
```

Comment:
Just a thought but for convenience what about accepting an ISO 8601 timestamp and converting that to a long?

Reply:
Yeah, we can implement a similar path that Oliver did in https://github.com/stargate/stargate/pull/1022/files#diff-afd1e86c5d7ba50a9f13687223a01d326662d1cfe10432a18e7fd242af038bd0R125-R150.
However, we need to support here: `BigInteger`, `Integer` and I wonder if also adding the `ISO 8601` won't be too wide and confusing. Maybe support for both `BigInteger` and `Integer` is good enough?

Reply:
I know I'm contradicting what I said through another channel, but maybe it's not that complicated after all. If we standardize on the format of [DateTimeFormatter.ISO_ZONED_DATE_TIME](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_ZONED_DATE_TIME), we can do it like this:

```java
  private static long parseTimestamp(String spec) {
    ZonedDateTime dateTime = ZonedDateTime.parse(spec);
    return dateTime.toEpochSecond() * 1_000_000 + dateTime.getNano() / 1000;
  }
```